File: python_cpp/uPIMulator_frontend/src/abi/isa/cc/const_cc_geu.py
from typing import Set

from abi.isa.cc._base_cc import BaseCC
from abi.isa.instruction.condition import Condition


class ConstCCGEU(BaseCC):
    def __init__(self, condition: Condition):
        super().__init__(ConstCCGEU.conditions(), condition)

    @staticmethod
    def conditions() -> Set[Condition]:
        return {Condition.GEU}

File: golang_vm/uPIMulator/src/device/linker/kernel/instruction/cc/const_cc_geu.go
package cc

import (
	"errors"
)

type ConstCcGeu struct {
	condition Condition
}

func (this *ConstCcGeu) Init(condition Condition) {
	conditions := map[Condition]bool{
		GEU: true,
	}

	if _, found := conditions[condition]; !found {
		err := errors.New("condition is not allowed")
		panic(err)
	}

	this.condition = condition
}

func (this *ConstCcGeu) Condition() Condition {
	return this.condition
}
